specification/app/resource-manager/Microsoft.App/stable/2025-07-01/CommonDefinitions.json

```diff
@@ -1282,6 +1300,193 @@
+        "name": "PrivateEndpointServiceConnectionStatus",
+        "modelAsString": true
+      }
+    },
+    "PrivateEndpointConnectionProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The current provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Waiting",
+        "Updating",
+        "Deleting",
+        "Pending"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointConnectionProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "PrivateLinkResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "A private link resource"
+    },
+    "PrivateLinkResourceProperties": {
+      "type": "object",
+      "properties": {
+        "groupId": {
+          "description": "The private link resource group id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requiredMembers": {
+          "description": "The private link resource required member names.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "requiredZoneNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The private link resource private link DNS zone name."
+        }
+      },
+      "description": "Properties of a private link resource."
+    },
+    "PrivateEndpointConnectionListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private endpoint connections",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "URL to get the next set of operation list results (if there are any).",
+          "readOnly": true
+        }
+      },
+      "description": "List of private endpoint connection associated with the specified resource"
+    },
+    "PrivateLinkResourceListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private link resources",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "URL to get the next set of operation list results (if there are any).",
```

Comment:
[ARMBlockingComment] Don't define these yourself. Use the definitions from common types
[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/privatelinks.json)

Reply:
We have differences in the globally defined definitions and our definitions and hence can't use them:


For example:



For Enum PrivateEndpointConnectionProvisioningState, we have defined a lot of values: https://github.com/Azure/azure-rest-api-specs/blob/3f52c24859ee4e05170b142fecdf463f5163b39d/specification/app/resource-manager/Microsoft.App/stable/2025-07-01/CommonDefinitions.json#L1392-L1403




Whereas the global privateLinks resource schema only has few values: https://github.com/Azure/azure-rest-api-specs/blob/be190c66354520d7b7028bf2be3150f311733a04/specification/common-types/resource-management/v5/privatelinks.json#L78-L85








This will result in breaking change and even need code changes in our Service layer.

Reply:
@TimLovellSmith

Reply:
Thats a fair comment. It makes me think the common types definition should probably include a bigger set of values, such as adding 'Updating', and 'Deleting' - those are definitely popular provisioning states that a lot of resources use. (Some resources also use 'Accepted' or 'Pending' but these are less common)

I am curious, do you really use all these states like 'Waiting'? Could you be okay with just Pending, Updating, or Accepted, or even just Creating and Updating?

I think while a number of resources use those states, 'Waiting' is actually pretty uncommon, and I'd just like to understand if its really needed/used.